runtime/include/mem/jemalloc/chpl-mem-impl.h

```diff
@@ -29,6 +29,11 @@
 
 #define MALLOCX_NO_FLAGS 0
 
+
+extern bool chpl_mem_impl_allocLocalizes;
+#define CHPL_MEM_IMPL_ALLOCLOCALIZES chpl_mem_impl_allocLocalizes
```

Comment:
I'd prefer not to have to mention it at all in the layers that can't do it.  We have a long-standing technique for doing optional stuff in layer implementations; see the discussion of `CHPL_TASK_func_IMPL_DECL` in chpl-tasks.h.  But that's come to seem clunky to me because the #ifdef controls whether the base .h file or the one from the implementation provides both the declaration and the definition.  So here what I was trying to do was to have the base .h file always provide the declaration, with the implementation optionally providing the implementation, effectively.